### ISSUE TITLE
Che 5.2.2 Installation fails on Docker 1.13.0

### ISSUE THREAD

**Issue by MuraliVidhya** [eclipse-che/che#4071]
Hi,

Eclipse Che Installation on Docker 1.13.0 fails with the below errors:

INFO: (che cli) : 5.2.2 - using docker 1.13.0 / native
/scripts/base/library.sh: line 555: [: 10: unary operator expected
/scripts/base/library.sh: line 308: 2 : unbound variable

**Reproduction Steps:**
docker run -it --rm -v /usr/local/codenvy/chedata:/data -v /var/run/docker.sock:/var/run/docker.sock eclipse/che start

**OS and version:**    
RHEL 7.3

**Comment by ghost** [comment]
@mvidhya86 can you remove all images and start anew?

`docker rmi -f $(docker ps -aq)`

**Comment by ghost** [comment]
@mvidhya86 any updates please?

**Comment by silverjam** [comment]
I'm seeing this issue too.

Did a clean install.  Output from che looks like this:

```
docker run -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v $PWD:/data eclipse/che start
Unable to find image 'eclipse/che:latest' locally
latest: Pulling from eclipse/che
b7f33cc0b48e: Pull complete
cd5da33c6280: Pull complete
5f6840fd6369: Pull complete
c4a35236e7f0: Pull complete
551e94661395: Pull complete
1572e5528426: Pull complete
7ecebaef7f59: Pull complete
81437276f6c4: Pull complete
Digest: sha256:93b199dc63e8dac73c4aaa3cf6ac5e6e09daede329f22b09406f5882c9f7f8d9
Status: Downloaded newer image for eclipse/che:latest
WARN: Bound 'eclipse/che' to 'eclipse/che:5.3.1'
WARNING: No swap limit support
INFO: Proxy: HTTP_PROXY=http://pitc-zscaler-americas-alpharetta3pr.proxy.corporate.ge.com:80, HTTPS_PROXY=http://pitc-zscaler-americas-alpharetta3pr.proxy.corporate.ge.com:80, NO_PROXY=localhost,127.0.0.1,.ge.com
INFO: (che cli): Pulling image alpine:3.4
INFO: (che cli): Pulling image eclipse/che-ip:5.3.1
INFO: (che cli): 5.3.1 - using docker 1.13.1 / native
/scripts/base/startup_04_pre_cli_init.sh: line 263: [: 10: unary operator expected
/scripts/base/startup_04_pre_cli_init.sh: line 383: 2: unbound variable
```

**Comment by silverjam** [comment]
I don't think this is a problem with the docker version (I tried downgrading to 1.12.3), but a proxy issue (for me at least), this line using the output from a curl command:

https://github.com/eclipse/che/blob/dcf6c63683f2431c601bc23bf4a8d5502e14d860/dockerfiles/base/scripts/base/startup_04_pre_cli_init.sh#L258

If the curl command files the string is empty, thus the error from the [ command.

I'm not able to work around the issue though.  I tried the following to no avail:

```
docker run --env http_proxy=$http_proxy \
    --env https_proxy=$https_proxy --env no_proxy=$no_proxy \
    -it --rm -v /var/run/docker.sock:/var/run/docker.sock -v $PWD:/data eclipse/che start
```

**Comment by silverjam** [comment]
@eivantsov I provided some additional info, thanks!

**Comment by TylerJewell** [comment]
Hmmm.  We had another account run into a similar issue and we added a detection and fix for 5.4 releasing shortly.  Does this issue happen if you use our nightly version?  

Also if you --skip:network does it happen?

**Comment by silverjam** [comment]
Using --skip:network works with the nightly and the 5.3 release.

**Comment by silverjam** [comment]
There are proxy detection routines in other startup_\*.sh scripts, I think they just need to move to the library.sh and get invoked in the startup_04\*.sh.

**Comment by TylerJewell** [comment]
In the 5.3 or 5.3 train, I cannot remember which one, we reorganized all of the bootstrap scripts so that the functions are loaded exactly when they are needed and not before.  I think you ran into some issues becuase you had some older (5.2.2) images on your disk.  In 5.4 we also introduced a hard binding of an API version to make sure that the files that you have in the Docker images that you are running match exactly any other files that you may be loading with the CLI image.  For example, our developers mount `:/repo` and will instead use scrpit files from their local git repo instead of the ones in the Docker image.  If the git repo has a branch opened and doesn't match what is in the CLI image, then weird things can happen.  So we now do hard checks for all of this stuff and print warnings or just error out.

Sorry you ran into this issue, but we just released and shipped 5.4 this morning.  So if you wipe everything and reload, these issues should no longer appear!
